### PR TITLE
Update the specialize/inline tactic

### PR DESCRIPTION
**Pull request by sonmarcho**
This PR makes minor improvements to the specialize/inline tactic (`code/meta`), and adds small examples for this tactic (`code/meta/example`).

### CODE REVIEW

File: code/meta/Meta.Interface.fst [hunk 1]
```diff
@@ -547,14 +556,19 @@ and visit_body (t_i: term)
           st, e, bvs, ses
       end
 
-  | Tv_Var _ | Tv_BVar _ | Tv_FVar _ | Tv_UInst _ _
+  | Tv_Var _ | Tv_BVar _ | Tv_UInst _ _ | Tv_FVar _
   | Tv_Const _ ->
       st, e, bvs, []
 
   | Tv_Abs b e ->
+      (* BETA: arguments of functions are allowed to refer to types to be
```
Comment (msprotz) [inline code/meta/Meta.Interface.fst:564]: need to remove this

File: code/meta/Meta.Interface.fst [hunk 2]
```diff
@@ -547,14 +556,19 @@ and visit_body (t_i: term)
...
+         specialized over, assuming no higher-order types *)
+      // let bv, _ = inspect_binder b in
+      // let { bv_sort = t } = inspect_bv bv in
+      // let st, _, bvs, ses0 = visit_body t_i index_bv inv_bv st bvs t in
       let st, e, bvs, ses = visit_body t_i index_bv inv_bv st bvs e in
       let e = pack (Tv_Abs b e) in
-      st, e, bvs, ses
+      st, e, bvs, ses // @ ses0
```
Comment (msprotz) [inline code/meta/Meta.Interface.fst:571]: same

**Review by msprotz — APPROVED** [review]
modulo changes, looks good!! thanks

File: code/meta/Meta.Interface.fst [hunk 3]
```diff
@@ -449,6 +457,7 @@ and visit_body (t_i: term)
   //   through as well to avoid allocating the same thing twice
   // ses is strictly bottom-up
   match inspect e with
+  // | Tv_FVar _
```
Comment (R1kM) [inline code/meta/Meta.Interface.fst:460]: Remove this

**Review by R1kM — APPROVED** [review]
Looks good to me too, modulo one extra cleanup. @sonmarcho , you can also drop the changes to INFO.txt to avoid conflicts.

**Pull request merged**